docs/timezone-draft.md

```diff
@@ -0,0 +1,180 @@
+# Draft design of custom time zone API
+
+## Why?
+
+Currently, `Temporal.TimeZone` objects can only be obtained by passing an identifier to the constructor or to `Temporal.TimeZone.from()`.
+These identifiers can be:
+
+- an offset string such as `-07:00`, in which case a time zone object is returned with a constant offset, with no daylight saving changes ever;
+- an IANA time zone name such as `America/New_York`, in which case a time zone object is returned that will give a different offset depending on the date that it is queried with, which attempts to be historically accurate (within reason) as to the actual time that clocks showed there.
+
+No other identifiers than the ones deemed valid by one of the above two rules (`±HH:MM` or [IsValidTimeZoneName](https://tc39.es/ecma402/#sec-isvalidtimezonename)), are currently permitted.
+
+Custom time zones are a desirable feature because implementations may want to prevent leaking information about what historical revision of the time zone data is present on the host system, or control what information about the host system's current time zone is exposed.
+See [issue #273](https://github.com/tc39/proposal-temporal/issues/273).
+
+API users may also want to perform time zone calculations with older, or newer, or more accurate versions of the time zone data than what is present in their implementation.
+
+As an example, the `America/Chicago` time zone in the tzdata database is based on the historical time in Chicago.
+Before the year 1883, when the US introduced time zones due to the influence of railroads, Chicago observed the local mean time of GMT-05:50:36.
+Other cities in what is now the `America/Chicago` time zone would have observed their own local mean times.
+If a user wanted to do accurate calculations for times occurring before 1883 in places in the `America/Chicago` time zone other than Chicago, a custom time zone would be required.
+For example, St. Louis's local mean time is GMT-06:00:49.11 ([Byrd, Mary E., 1899, _A Laboratory Manual In Astronomy_, p. 63. Athenaeum Press: Boston.](https://books.google.ca/books?id=Xfg3AAAAMAAJ&pg=PA63))
+
+## Usage of a custom time zone
+
+Like built-in time zones, custom time zones have an `[[Identifier]]`.
+In these examples we assume a custom time zone class `StLouisTime` with the identifier `America/St_Louis`.
+(See the following section for how such a time zone object would be obtained.)
+
+When parsing an ISO 8601 string, the only places the time zone identifier is taken into account are `Temporal.Absolute.from()` and `Temporal.TimeZone.from()`.
+
+These functions gain an `options` parameter with an `idToTimeZone` option, whose value is a function returning the appropriate time zone object for the `id` passed in, or `null` to indicate that the time zone doesn't exist.
+
+If the option is not given, `Temporal.TimeZone.idToTimeZone()` is called.
```

Comment:
If it wasn't for the fact that it makes sense to match the `idToTimeZone` option, I'd say `Temporal.TimeZone.fromId()` would make more sense. Now I don't know. On the other hand, do we need this given that `Temporal.TimeZone.from()` exists?

Reply:
I think we can hash this out in #294.